File: leetcode/linked_list/1474_Delete_N_Nodes_After_M_Nodes_of_a_Linked_List/codes/ll.c
/**
 * Definition for singly-linked list.
 */

struct ListNode {
    int val;
    struct ListNode *next;
};
 
struct ListNode* deleteNodes(struct ListNode* head, int m, int n) {
    struct ListNode* curr = head;
    while (curr != NULL) {
        for (int i = 1; i < m && curr != NULL; i++) {
            curr = curr -> next;
        }
        if (curr == NULL) break;

        struct ListNode* temp = curr -> next;
        for (int i = 0; i < n && temp != NULL; i++) {
            struct ListNode* toDelete = temp;
            temp = temp -> next;
        }
        curr -> next = temp;
        curr = temp;
    }
    return head;
}

File: leetcode/linked_list/1474_Delete_N_Nodes_After_M_Nodes_of_a_Linked_List/codes/ll.py
from typing import Optional

# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def deleteNodes(self, head: Optional[ListNode], m: int, n: int) -> Optional[ListNode]:
        curr = head
        while curr:
            for _ in range(1, m):
                if curr:
                    curr = curr.next
            if not curr:
                break

            temp = curr.next
            for _ in range(n):
                if temp:
                    temp = temp.next
                else:
                    break
            curr.next = temp
            curr = temp
        return head 